### PR TITLE
[EASI 2429] Operational Needs, Operational Solution Access control

### PR DESCRIPTION
**Pull request by rileyanderson**
# EASI-2429

## Changes and Description

- Add OperationalNeedRelation
- Enable/Update SolutionRelation

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by StevenWadeOddball — COMMENTED** [review]
something to note, document solution link should probably use the solution link relation

**Comment by rileyanderson** [comment]
> something to note, document solution link should probably use the solution link relation

@StevenWadeOddball [It does](https://github.com/CMSgov/mint-app/blob/main/pkg/models/plan_document_solution_link.go#L8). It was already there 😄

File: pkg/models/operational_solution.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 // OperationalSolution represents the specific implemented solution to an Operational Need
 type OperationalSolution struct {
 	baseStruct
+	operationalNeedRelation
 	OperationalNeedID uuid.UUID               `json:"operationalNeedID" db:"operational_need_id"`
```
Comment (StevenWadeOddball) [inline pkg/models/operational_solution.go:13]: This should be removed, as the OperationalNeedID field comes from the embedded struct already, so this is redundant.

**Review by StevenWadeOddball — CHANGES_REQUESTED** [review]
Overall looks pretty good! There are a few places where pre-update still need to be enabled so access control works ( EG Operational_Solution.go (resolvers), the call to BaseStruct Preupdate should provide true instead of false for check access. 

I will make these updates.

File: pkg/models/operational_solution.go (hunk 1 above)
Comment (StevenWadeOddball) [reply pkg/models/operational_solution.go:13]: When you run the code, one of the variables stays null unless you set it 
image

Comment (StevenWadeOddball) [reply pkg/models/operational_solution.go:13]: This is corrected removing the extraneous field
image

Comment (StevenWadeOddball) [reply pkg/models/operational_solution.go]: I resolved this

**Comment by StevenWadeOddball** [comment]
@OddTomBrooks and @ClayBenson94 , can you please review this when you have a chance? I made enough changes that I think it should get approval from you two instead of me. It's ready for review.

**Review by ClayBenson94 — APPROVED** [review]

**Review by StevenWadeOddball — APPROVED** [review]

**Pull request merged**